### DISCUSSION TITLE
Opening files with Onlyoffice

### DISCUSSION THREAD

**Discussion by euglevi** [sxyazi/yazi#1856] [1. Q&A]
### What system are you running Yazi on?

Linux Wayland

### What terminal are you running Yazi in?

foot v1.18.1

### `yazi --debug` output

```Shell
Yazi
    Version: 0.3.3 (5eabd7d 2024-10-29)
    Debug  : false
    OS     : linux-x86_64 (unix)

Ya
    Version: 0.3.3 (VERGEN_IDEMPOTENT_OUTPUT 2024-09-18)

Emulator
    Emulator.via_env: ("foot", "")
    Emulator.via_csi: Ok(Foot)
    Emulator.detect : Foot

Adapter
    Adapter.matches: Sixel

Desktop
    XDG_SESSION_TYPE           : Some("wayland")
    WAYLAND_DISPLAY            : Some("wayland-1")
    DISPLAY                    : Some(":0")
    SWAYSOCK                   : Some("/run/user/1000/sway-ipc.1000.2228.sock")
    HYPRLAND_INSTANCE_SIGNATURE: None
    WAYFIRE_SOCKET             : None

SSH
    shared.in_ssh_connection: false

WSL
    WSL: false

Variables
    SHELL              : Some("/usr/bin/zsh")
    EDITOR             : Some("nvim")
    VISUAL             : None
    YAZI_FILE_ONE      : None
    YAZI_CONFIG_HOME   : None

Text Opener
    default     : Some(Opener { run: "${EDITOR:-vi} \"$@\"", block: true, orphan: false, desc: "$EDITOR", for_: None, spread: true })
    block-create: Some(Opener { run: "${EDITOR:-vi} \"$@\"", block: true, orphan: false, desc: "$EDITOR", for_: None, spread: true })
    block-rename: Some(Opener { run: "${EDITOR:-vi} \"$@\"", block: true, orphan: false, desc: "$EDITOR", for_: None, spread: true })

Multiplexers
    TMUX               : false
    tmux version       : No such file or directory (os error 2)
    tmux build flags   : enable-sixel=Unknown
    ZELLIJ_SESSION_NAME: None
    Zellij version     : Permission denied (os error 13)

Dependencies
    file             : 5.45
    ueberzugpp       : No such file or directory (os error 2)
    ffmpegthumbnailer: 2.2.3
    magick           : 7.1.1-39
    fzf              : 0.55
    fd               : No such file or directory (os error 2)
    rg               : 14.1.1
    chafa            : No such file or directory (os error 2)
    zoxide           : 0.9.4
    7z               : 24.08
    7zz              : No such file or directory (os error 2)
    jq               : 1.7
```

### Describe the bug

First of all, thank you for this wonderful file manager!

I have a small problem. When I try to open a docx or xlsx file with Onlyoffice it takes forever, and sometimes it fails. It opens a window, but that window is either empty and non-responsive or with the file but still non-responsive. This does not happen to me, for instance, with nnn or with any other way of opening files with Onlyoffice.

### Minimal reproducer

Install Onlyoffice, last version, open yazi and select any Microsoft Office file.

### Validations

- [X] I tried the [latest nightly build](https://yazi-rs.github.io/docs/installation#official-binaries), and the issue is still reproducible
- [X] I updated the debug information (`yazi --debug`) input box to the nightly that I tried
- [X] I can reproduce it after disabling all custom configs/plugins (`mv ~/.config/yazi ~/.config/yazi-backup`)

**Comment by sxyazi** [comment]
I assume the opener you're using is `xdg-open` since you're using the default config:

https://github.com/sxyazi/yazi/blob/5eabd7d89766d41e8c5bafc841c74b2b23ed30ae/yazi-config/preset/yazi.toml#L38-L42

Could you please try running it directly on these files in the terminal to see if it works?

**Reply by euglevi** [reply]
It works. I also have a script with rofi that uses xdg-open and I never had any issues with that. 

I made a couple of experiments to see what was the problem. More specifically:
a) I tried with WPS Office. The same problem arise. This makes me think that it is an issue related more in general to xwayland and not specific of Onlyoffice.
b) the problem is still there in Hyprland.
b) the problem does not arise in wayland Gnome, so this must be specific to wlroots-based compositors.

Thank you for your effort (and the great file manager!).

**Reply by sxyazi** [reply]
What if you give the `xdg-open` a [`orphan = true`](https://yazi-rs.github.io/docs/configuration/yazi/#opener), i.e.

 ```
{ run = 'xdg-open "$1"', desc = "Open", orphan = true }
```

**Reply by euglevi** [reply]
I have tried it, but the problem is still there.

**Reply by sxyazi** [reply]
That's weird, maybe the ultimate way is to call Onlyoffice directly instead of using `xdg-open` to launch it

**Reply by euglevi** [reply]
Good point. I added a new opener: 

```
onlyoffice =[
	{ run = 'onlyoffice-desktopeditors "$1"',    orphan = true, for = "unix" },
]
```
and a new rule:

```
prepend_rules = [
	{ name = "*", use = [ "onlyoffice", "open" ] },
}
```
No luck. So I guess we can discard that this is a problem with xdg-open.

**Reply by euglevi** [reply]
Just to follow up a little bit on this (not sure if I am the only one experiencing this issue...), this happens to me with other terminals too. For instance, I was trying `kitty` and I had the same issue.

**Reply by euglevi** [reply]
It seems that the issue is now solved after wlroots updated to 0.18.1. Shall we close the discussion?

**Reply by sxyazi** [reply]
Nice to see you've found the cause of the problem and got it solved! Closing this now
